libs/qcustomplot-source/qcustomplot.cpp

```diff
@@ -25030,7 +25030,7 @@ QCPColorMapData &QCPColorMapData::operator=(const QCPColorMapData &other)
     if (!isEmpty())
     {
       memcpy(mData, other.mData, sizeof(mData[0])*keySize*valueSize);
-      if (mAlpha)
+      if (other.mAlpha && mAlpha)
```

Comment:
Shouldn't this be requested upstream in the [qcustomplot library](https://gitlab.com/DerManu/QCustomPlot), so it is not lost when we update to a new release?

Reply:
Could we keep this *and* request a merge upstream? As it looks a bit risky. IIRC, the behaviour is undefined when passing a `nullptr` as the source to `memcpy`.

Reply:
https://gitlab.com/revolter/QCustomPlot/merge_requests/1/

Reply:
Yeah, the general approach (where practical) is to submit a PR/MR/etc upstream first and get it accepted there.  Then, whenever that's finished (it can sometimes take a few revisions), merge the same (final, accepted) change into our source too.

The reason for doing it upstream first, is so that if the initial intended patch needs a few revisions then we're not still using an older version of the patch from before it was accepted. :smile: